generators/exercises/list_ops.py

```diff
@@ -0,0 +1,124 @@
+FUNC_PROTO = """\
+#include "vendor/unity.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define BUFFER_SIZE 16
+
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+
+extern size_t append(int64_t *buffer, const int64_t *list1, size_t list1_count, const int64_t *list2, size_t list2_count);
+extern size_t filter(int64_t *buffer, const int64_t *list, size_t list_count, bool (*function)(int64_t));
+extern size_t map(int64_t *buffer, const int64_t *list, size_t list_count, int64_t (*function)(int64_t));
+extern int64_t foldl(const int64_t *list, size_t list_count, int64_t initial, int64_t (*function)(int64_t, int64_t));
+extern int64_t foldr(const int64_t *list, size_t list_count, int64_t initial, int64_t (*function)(int64_t, int64_t));
+extern size_t reverse(int64_t *buffer, const int64_t *list, size_t list_count);
+
+static bool is_odd(int64_t x) {
+    return x % 2 == 1;
+}
+
+static int64_t plus_one(int64_t x) {
+    return x + 1;
+}
+
+static int64_t multiply(int64_t acc, int64_t el) {
+    return el * acc;
+}
+
+static int64_t add(int64_t acc, int64_t el) {
+    return el + acc;
+}
+
+static int64_t divide(int64_t acc, int64_t el) {
+    return acc == 0 ? 0 : el / acc;
+}
+"""
+
+def array_literal(digits):
```

Comment:
Those are not necessarily digits, right? Maybe call it `values` or `numbers` instead?